application/controllers/UserManagementController.php

```diff
@@ -1319,20 +1319,10 @@ public function loadModel(int $id): User
     public function updateAdminUser(array $aUser): User
     {
         $oUser = User::model()->findByPk($aUser['uid']);
-        //If the user id of the post is spoofed somehow it would be possible to edit superadmin users
-        //Therefore we need to make sure no non-superadmin can modify superadmin accounts
-        //Since this should NEVER be the case without hacking the software, this will silently just do nothing.
-        if (
-            !Permission::model()->hasGlobalPermission('superadmin', 'read', Yii::app()->user->id)
-            && Permission::model()->hasGlobalPermission('superadmin', 'read', $oUser->uid)
-        ) {
-            throw new CException("This action is not allowed, and should never happen", 500);
-        }
-
         // Abort if logged in user has no access to this user.
         // Using same logic as User::getButtons().
         $userManager = new UserManager(Yii::app()->user, $oUser);
-        if (!$userManager->canEdit()) {
+        if (!$userManager->canEdit() || $aUser['uid'] == Yii::app()->user->id) {
```

Comment:
This is not the place to edit self.

Reply:
This rule should be inside the userManager->canEdit() ?

Reply:
Actually, I am seeing below, isn't there already? Then, is this necesarry?

Reply:
> This rule should be inside the userManager->canEdit() ?

Before userManager->canEdit()  allow $aUser['uid'] == Yii::app()->user->id


> Actually, I am seeing below, isn't there already? Then, is this necesarry?

No canEdit return true, but  _This is not the place to edit self._

Reply:
> This rule should be inside the userManager->canEdit() ?

1. [userManager->canEdit()](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/services/UserManager.php#L82) **allow** user to edit himself
2. UserManagement [must disallow user to edit himself](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/User.php#L584)  it's the task of personalsettings not usermanagement.

We can allow it, but currently : we hide the button. Let @tiborpacalat choose maybe ? 
_Did user can edit himself in user management._

Reply:
> [userManager->canEdit()](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/services/UserManager.php#L82) allow user to edit himself
UserManagement [must disallow user to edit himself](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/User.php#L584) it's the task of personalsettings not usermanagement.

Oh, I think I got it. Then I think good.

Reply:
Why don't we add these exact comments in the code?

> [userManager->canEdit()](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/services/UserManager.php#L82) allow user to edit himself
UserManagement [must disallow user to edit himself](https://github.com/LimeSurvey/LimeSurvey/blob/25f63d5758134ad5eca607453deb8a9e94b7f2bd/application/models/User.php#L584) it's the task of personalsettings not usermanagement.